src/target/renesas.c

```diff
@@ -514,13 +632,33 @@ static bool renesas_rv40_done(target_flash_s *const f)
 	return renesas_rv40_pe_mode(t, PE_MODE_READ);
 }
 
+static bool renesas_check_option_setting(target_s *t, target_addr_t addr)
+{
+	renesas_priv_s *priv_storage = (renesas_priv_s *)t->target_storage;
+	if (!priv_storage)
+		return false;
+
+	if ((priv_storage->details.option_start != 0 && addr >= priv_storage->details.option_start &&
+			addr < (priv_storage->details.option_start + priv_storage->details.option_size)) ||
+		(priv_storage->details.option_start_2 != 0 && addr >= priv_storage->details.option_start_2 &&
+			addr < (priv_storage->details.option_start_2 + priv_storage->details.option_size_2)))
+		return true;
+	else
+		return false;
+}
+
 /* !TODO: implement blank check */
 static bool renesas_rv40_flash_erase(target_flash_s *f, target_addr_t addr, size_t len)
 {
 	target_s *const t = f->t;
 
+	uint16_t block_size;
+
 	/* Code flash or data flash operation */
-	const bool code_flash = addr < RENESAS_CF_END;
+	if (addr < RENESAS_CF_END)
+		block_size = addr < RV40_CF_REGION0_SIZE ? RV40_CF_REGION0_BLOCK_SIZE : RV40_CF_REGION1_BLOCK_SIZE;
+	else
+		block_size = f->blocksize;
```

Comment:
This looks like it's worth lifting into its own function so  you can say `const uint16_t block_size = renesas_rv40_block_size(f->blocksize, addr);` or so - this will improve codegen and `const`-correctness